services/api-server/src/simcore_service_api_server/api/routes/studies_jobs.py

```diff
@@ -231,15 +233,23 @@ async def inspect_study_job(
     "/{study_id}/jobs/{job_id}/outputs",
     response_model=JobOutputs,
     include_in_schema=API_SERVER_DEV_FEATURES_ENABLED,
-    status_code=status.HTTP_501_NOT_IMPLEMENTED,
-    response_description="Not implemented",
 )
 async def get_study_job_outputs(
     study_id: StudyID,
     job_id: JobID,
+    user_id: Annotated[PositiveInt, Depends(get_current_user_id)],
+    webserver_api: Annotated[AuthSession, Depends(get_webserver_session)],
+    storage_client: Annotated[StorageApi, Depends(get_api_client(StorageApi))],
 ):
-    msg = f"get study job outputs study_id={study_id!r} job_id={job_id!r}. SEE https://github.com/ITISFoundation/osparc-simcore/issues/4177"
-    raise NotImplementedError(msg)
+    job_name = _compose_job_resource_name(study_id, job_id)
+    _logger.debug("Getting Job Outputs for '%s'", job_name)
+
+    project_outputs = await webserver_api.get_project_outputs(job_id)
```

Comment:
I have some doubts about your approach here. 

Somehow what you do is to check whether the backend does it right (you check that the project name is well defined upon construction), and if there is an inconsistency you "blame" the client  by responding with a 4XX :-D

Reply:
Hmm, I definitely agree this is not the right way to do it, but it was the best way I could see to do it. What I would like to check is that the `study_id` is the "parent" of the `job_id`. Currently the only way I can do this is by checking the name is what I expect it to be. So, to put it differently, I trust the backend and if the name is not correct I believe it is because the `study_id` is not the parent of the `job_id`.

Reply:
aha I see now your point. You want to prevent the client to hijack the interface to get info on non-job projects, right?